src/runtime/float.cpp

```diff
@@ -998,6 +1058,47 @@ Box* floatTrunc(BoxedFloat* self) {
     return PyLong_FromDouble(wholepart);
 }
 
+Box* floatTrunc(BoxedFloat* self) {
+    if (!PyFloat_Check(self))
+        raiseExcHelper(TypeError, "descriptor '__trunc__' requires a 'float' object but received a '%s'",
+                       getTypeName(self));
+
+    return floatToInt(self);
+}
+
+Box* floatInt(BoxedFloat* self) {
+    if (!PyFloat_Check(self))
+        raiseExcHelper(TypeError, "descriptor '__int__' requires a 'float' object but received a '%s'",
+                       getTypeName(self));
+    return floatToInt(self);
+}
+
+Box* floatLong(BoxedFloat* self) {
+    if (!PyFloat_Check(self))
+        raiseExcHelper(TypeError, "descriptor '__long__' requires a 'float' object but received a '%s'",
+                       getTypeName(self));
+
+    double wholepart; /* integral portion of x, rounded toward 0 */
+
+    (void)modf(self->d, &wholepart);
```

Comment:
this and the int version will fail for inf floats
```>>float("inf").__int__()
pyston_dbg: ../../src/runtime/objmodel.cpp:4158: pyston::Box *pyston::callCLFunc(pyston::FunctionMetadata *, pyston::CallRewriteArgs *, int, pyston::BoxedClosure *, pyston::BoxedGenerator *, pyston::Box *, pyston::Box *, pyston::Box *, pyston::Box *, pyston::Box **) [S = 1, rewritable = 0]: Assertion `S == CAPI' failed.
```

And I think we can remove this ```(void)modf(self->d, &wholepart);``` call (at least cpython does not do this AFAIK)